File: 图灵/爬虫15期进阶/逆向课/day5-请求和JS调用/py测试.py
# -*- coding: utf-8 -*-


import execjs


with open("py测试.js", encoding='utf-8') as f1:
    js_code = f1.read()

ctll = execjs.compile(js_code)
# 调用
res = ctll.call("get")

# 传参
res1 = ctll.call("getPar","how are you")

res2 = ctll.call("getPar1","how are you","asdasd",{"page":"111"})

res3 = ctll.call("getPars",{"page":"111"})

print(res2)

print(execjs.eval('Date.now()'))

File: 图灵/爬虫15期进阶/逆向课/day5-请求和JS调用/py测试.js


function get(){
    return "hello world"
}


function getPar(arg){
    return "hello world" + arg
}

function getPar1(arg,aaa,bbb){
    let pages = bbb['page']
    return aaa + "hello world" + arg + pages
}

function getPars(obj){
    let page = obj['page']
    console.log(page)
    return page
}




